lua/titan/lsp.lua

```diff
@@ -54,239 +39,169 @@ local capabilities = vim.lsp.protocol.make_client_capabilities()
-      },
-    }
-  )
-
-  return language_server_cmd(opts)
-end
-
-local function tsserver_cmd(opts)
-  opts = opts or {}
-  opts = vim.tbl_deep_extend(
-    "force",
-    opts,
-    {
-      name = "tsserver",
-      locations = {
-        ".bin/typescript-language-server",
-      },
-      fallback_locations = {
-        "node_modules/.bin/typescript-language-server",
-        "typescript-language-server",
-      },
-    }
-  )
-
-  return language_server_cmd(opts)
-end
-
-M.custom_commands = nil
-
-function M.add_custom_command(name, cmd)
-  M.custom_commands[name] = cmd
 end
 
-function M.reload_custom_commands()
-  logger.debug("Reload custom commands: ", M.custom_commands)
-  for name, cmd in pairs(M.custom_commands) do
-    logger.debug("Reloading ", name, cmd)
+local function lsp_cmd_override(server_name, opts, cmd_path, args)
+  args = args or {}
 
-    local config = lspconfig[name]
+  local cmd = build_command(server_name, cmd_path, args)
+  if cmd ~= nil then
+    opts.cmd = cmd
+  end
 
-    if config.manager then
-      for _, client in ipairs(config.manager.clients()) do
-        logger.debug("Stopping client", client.name)
-        client.stop(true)
-      end
+  opts.on_new_config = function(new_config, _)
+    local new_cmd = build_command(server_name, cmd_path, args)
+    if new_cmd ~= nil then
+      new_config.cmd = new_cmd
     end
   end
 end
 
 function M.setup()
-  M.custom_commands = {}
+  require("mason").setup()
+  mason_lspconfig.setup {
+    ensure_installed = {
+      'clangd',
+      'rust_analyzer',
+      'pyright',
+      'terraformls',
+      'bashls',
+      'elixirls',
+      'solargraph',
+      'tsserver',
+      'sumneko_lua',
+    },
+  }
 
   -- vim.lsp.set_log_level("trace")
   -- require("vim.lsp.log").set_format_func(vim.inspect)
 
-  M.add_custom_command("elixirls", function()
-    return { elixirls_cmd() }
-  end)
-  M.add_custom_command("solargraph", function()
-    return { solargraph_cmd(), "stdio" }
-  end)
-  M.add_custom_command("tsserver", function()
-    return { tsserver_cmd(), "--stdio" }
-  end)
-
-  local enhance_server_opts = {
-    ["elixirls"] = function(opts)
-      opts.cmd = M.custom_commands.elixirls()
-      opts.on_new_config = function(new_config, _)
-        new_config.cmd = M.custom_commands.elixirls()
-      end
+  mason_lspconfig.setup_handlers {
+    -- The first entry (without a key) will be the default handler
+    -- and will be called for each installed server that doesn't have
+    -- a dedicated handler.
+    function (server_name) -- default handler (optional)
+      local opts = {
+        on_attach = on_attach,
+        capabilities = capabilities,
+      }
+
+      lsp_setup(server_name, opts)
+    end,
+
+    ["elixirls"] = function(server_name)
+      local opts = {
+        on_attach = on_attach,
+        capabilities = capabilities,
+      }
+
+      lsp_cmd_override(
+        server_name,
+        opts,
+        ".elixir-ls-release/language_server.sh"
```

Comment:
yay! how will this affect subsequent commands given to language_server.sh in our repos? e.g., referencing that `debugger.sh`

Reply:
At the moment, this is completely focused on LSP stuff. DAP is the next PR!